docker/pytorch/Dockerfile

```diff
@@ -1,6 +1,99 @@
-FROM mosaicml/python:latest
+ARG BASE_IMAGE
+
+FROM ${BASE_IMAGE}
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        libgomp1 \
+        curl \
+        build-essential \
+        software-properties-common \
+        # For PILLOW:
+        zlib1g-dev \
+        libtiff-dev \
+        libfreetype6-dev \
+        liblcms2-dev \
+        tcl \
+        libjpeg8-dev && \
+    apt-get autoclean && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# # Upgrade NCCL to fix NVLink errors
+# # No Need to install, as the cuda 11.3 image already contains a late enough version
+# RUN DISTRO="$(lsb_release -si | tr '[:upper:]' '[:lower:]')$(lsb_release -sr | sed 's/\.//')" &&  \
+#     apt-get update && \
+#     apt-get install -y --no-install-recommends \
+#         dirmngr \
+#         gpg-agent && \
+#     apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/${DISTRO}/x86_64/7fa2af80.pub && \
+#     add-apt-repository "deb https://developer.download.nvidia.com/compute/cuda/repos/${DISTRO}/x86_64/ /" && \
+#     apt-get install -y --no-install-recommends --allow-change-held-packages \
+#         libnccl2 \
+#         libnccl-dev && \
+#     apt-get autoclean && \
+#     apt-get clean && \
+#     rm -rf /var/lib/apt/lists/*
+
+# # Use system installed NCCL per update above, point to library
+# ENV USE_SYSTEM_NCCL=1
+# ENV LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libnccl.so.2.9.6
+
+##############################
+# Install NodeJS (for Pyright)
+##############################
+RUN \
+    curl -fsSL https://deb.nodesource.com/setup_17.x | bash - && \
+    apt-get install -y --no-install-recommends nodejs && \
+    apt-get autoclean && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+################
+# Install Python
+################
+ARG PYTHON_VERSION
+
+RUN add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+    python3-apt \
+    python${PYTHON_VERSION} \
+    python${PYTHON_VERSION}-dev \
+    python${PYTHON_VERSION}-distutils && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python${PYTHON_VERSION} 1 && \
+    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python${PYTHON_VERSION} 1 && \
+    apt-get autoclean && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN curl -fsSL https://bootstrap.pypa.io/get-pip.py | python -
+
+
+#################
+# Install Pytorch
+#################
+ARG PYTORCH_VERSION
+ARG TORCHVISION_VERSION
+ARG CUDA_VERSION_TAG
 
 RUN pip install --no-cache-dir --upgrade pip && \
+    pip install --no-cache-dir --upgrade pillow_simd==7.0.0.post3 && \
```

Comment:
Inconvenient but this needs to happen *after* uninstalling the torch default pillow.  Check out Installation Instructions here:
https://github.com/uploadcare/pillow-simd